TreeView Load Data on Demand: rewrite (#1002)
DevExpress/devextreme-demos: JSDemos/Demos/TreeView/LoadDataOnDemand/description.md

> If [virtual mode](/Demos/WidgetsGallery/Demo/TreeView/VirtualMode/) does not suit your scenario, you can use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to specify custom logic to load data. This function is called each time a user expands a node that has not been expanded before. The node's identifier is passed to the function as an argument. Send this identifier to your data service that should return data for child nodes.

Reviewer: >  Send this identifier to your data service that should return data for child nodes.

Why would I need to do this?

Author: The createChildren front-end function instructs the back-end service to expand a certain row. To do this, the back-end service must get the node's identifier.

Reviewer 2: `does not suit your scenario` Why would it not 'suit your scenario'?

Reviewer 2: - `that has not been expanded before` -> that was not expanded before
- `Send this identifier to your data service that should return data for child nodes` -> Send this identifier to the data service that should return data for child nodes

Author: > does not suit your scenario Why would it not 'suit your scenario'?

Virtual mode uses the default logic to load data on demand. If it is not what a user needs, they should use the createChildren function.

Reviewer 2: Perhaps just state: 

```suggestion
You can also use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to specify custom logic to load data. This function is called each time a user expands a node that was not expanded before. The node's identifier is passed to the function as an argument. Send this identifier to your data service that should return data for child nodes.
```

Author: Reference to virtual mode should be in the text. What about the following?
```suggestion
You can use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to specify custom logic to load data. This function is called each time a user expands a node that was not expanded before. The node's identifier is passed to the function as an argument. Send this identifier to the data service that should return data for child nodes.

As an alternative, you can enable [virtual mode](/Demos/WidgetsGallery/Demo/TreeView/VirtualMode/) to use the default load logic.
```

> Use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to load data on demand and enhance performance on large datasets. Unlike virtual mode, this function allows you to use your own custom logic. 

Reviewer 3: I would suggest to start the first paragraph with a sentence where you describe what TreeView can do. Also, it may be useful to highlight that it is called each time when a parent node is clicked.

> Use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to load data on demand and enhance performance on large datasets. Unlike virtual mode, this function allows you to use your own custom logic. 

Reviewer 4: ```suggestion
If [virtual mode](/Demos/WidgetsGallery/Demo/TreeView/VirtualMode/) does not suit your use case, you can use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to specify custom logic to load data. This function is called each time a user expands a node that has not been expanded before. The node's identifier is passed to the function as an argument. Send this identifier to your data service that should return data for child nodes.
```

> To specify the function, pass the expanded node's identifier as an argument and process the returned value inside the function. This value can be an array of objects or a promise as in this demo. The following prerequisites should be met:

Reviewer 3: >To specify the function,

You do not want to specify the function but enable some behavior, right?

>pass the expanded node's identifier as an argument and process the returned value inside the function.

When a function returns a value, the function's execution is over, isn't it (I really just wondering, I might be wrong)? How about the following:
To enable this behavior, pass the the expanded node's data to the **createChildren** function and process this data in it.

>The following prerequisites should be met:

I'm not sure that the word 'prerequisites' is the best choice. I would suggest keeping the sentence from the createChildren description: 'This function has the following restrictions:'

* Neither the [dataSource](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataSource), nor the items property should be specified.

* The dataset should have a plain structure and the [dataStructure](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataStructure) property should be set to *"plain"*.

> * The TreeView should not add the [search panel](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEnabled) because the search is performed on loaded nodes and may produce incorrect results.

Reviewer 3: It is a little bit weird to see that TreeView should or should not do something.  Maybe rephrase some of the list items in imperative mood?

- Do not specify the items or dataSource properties.
- The dataset should have a plain structure and the [dataStructure](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataStructure) property should be set to *"plain"*.
- We recommend that you do not [enable search](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEnabled) because the search is performed on loaded nodes and may produce incorrect results.

> To specify the function, pass the expanded node's identifier as an argument and process the returned value inside the function. This value can be an array of objects or a promise as in this demo. The following prerequisites should be met:
> 
> * Neither the [dataSource](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataSource), nor the items property should be specified.
> 
> * The dataset should have a plain structure and the [dataStructure](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataStructure) property should be set to *"plain"*.
> 
> * The TreeView should not add the [search panel](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEnabled) because the search is performed on loaded nodes and may produce incorrect results.

Reviewer 4: Customes can find the information about restrictions in the API reference.

```suggestion
```
